cmd/wick/helpers.go

```diff
@@ -125,22 +129,317 @@ func readFromProfile(profile string) (*core.ClientInfo, error) {
 	// FIXME: validate not empty
 	clientInfo.Authid = section.Key("authid").String()
 	clientInfo.Authrole = section.Key("authrole").String()
-	// FIXME: validate authmethod is not of invalid type
+
 	clientInfo.AuthMethod = section.Key("authmethod").String()
+	if err = validateAuthMethod(clientInfo.AuthMethod); err != nil {
+		return nil, err
+	}
 	if clientInfo.AuthMethod == cryptosignAuth {
-		// Validate private key not empty and is valid length
 		clientInfo.PrivateKey = section.Key("private-key").String()
+		if err = validatePrivateKey(clientInfo.PrivateKey); err != nil {
+			return nil, err
+		}
 	} else if clientInfo.AuthMethod == ticketAuth {
-		// validate ticket not empty
 		clientInfo.Ticket = section.Key("ticket").String()
+		if clientInfo.Ticket == "" {
+			return nil, fmt.Errorf("ticket is required for ticket authentication")
+		}
 	} else if clientInfo.AuthMethod == wampCraAuth {
-		// validate not empty
 		clientInfo.Secret = section.Key("secret").String()
+		if clientInfo.Secret == "" {
+			return nil, fmt.Errorf("secret is required for wampcra authentication")
+		}
 	}
 
 	return clientInfo, nil
 }
 
+// validateURL returns error if given string is not valid url.
+func validateURL(s string) error {
+	if s == "" {
+		return fmt.Errorf("invalid url : url must not be empty")
+	}
+	parse, err := url.ParseRequestURI(s)
+	if err != nil {
+		return err
+	}
+	switch parse.Scheme {
+	case "ws", "rs", "tcp":
+		return nil
+	default:
+		return fmt.Errorf("invalid url: scheme must be 'ws', 'rs' or 'tcp'")
+	}
+}
+
+// validateSerializer returns error if given string is not a valid serializer.
+func validateSerializer(s string) error {
+	switch s {
+	case jsonSerializer, msgpackSerializer, cborSerializer:
+		return nil
+	default:
+		return fmt.Errorf("invalid serializer: serailizer must be json, msgpack or cbor")
```

Comment:
to be consistent with the rest of the errors:
```suggestion
		return fmt.Errorf("invalid serializer: serailizer must be 'json', 'msgpack' or 'cbor'")
```

Reply:
Done